.github/CONTRIBUTING.md

```diff
@@ -1,32 +1,122 @@
+ - Take a look at the [Contribution Guidelines](https://checkstyle.org/contributing.html) for
+    on how to contribute to the project.
+ - Select an issue to work on from the
+    [Issues Page](https://github.com/checkstyle/checkstyle/issues). We
+    have a few issues labeled as
+    https://github.com/checkstyle/checkstyle/labels/good%20first%20issue to
+    help you get started. Once you get your first PR merged, you can move on to
+    https://github.com/checkstyle/checkstyle/labels/good%20second%20issue,
+    https://github.com/checkstyle/checkstyle/labels/good%20third%20issue,
+    and finally https://github.com/checkstyle/checkstyle/labels/good%20fourth%20issue
+    before taking up some bug fixes or features.
 
-Please follow the below links in order to start contributing to the checkstyle project
+## :book: Code of Conduct
 
-* [Beginning Development](https://checkstyle.org/beginning_development.html)
+This project and everyone participating in it is governed by the
+    [CheckStyle Code of Conduct](/.github/CODE_OF_CONDUCT.md).
 
-* [Contribution Guidelines](https://checkstyle.org/contributing.html)
+## :bulb: Asking Questions
 
-Find the links below to import the checkstyle project to an IDE
+See our [Discussions Page](https://github.com/checkstyle/checkstyle/discussions). In short, GitHub
+issues are not the appropriate place to debug your specific project, but should be reserved
+for filing bugs and feature requests.
 
-* [Importing checkstyle project to IntelliJ](https://checkstyle.org/idea.html)
+## :inbox_tray: Opening an Issue
 
-* [Importing checkstyle project to Eclipse](https://checkstyle.org/eclipse.html)
+Before [Opening an Issue](https://github.com/checkstyle/checkstyle/issues),
+check if you are using the latest version of the project,
+[found here](https://github.com/checkstyle/checkstyle/releases). If you are not up-to-date,
+see if updating to the latest release fixes your issue.
 
-* [Importing checkstyle project to Netbeans](https://checkstyle.org/netbeans.html)
+A great way to contribute to the project is to create a detailed issue when you encounter
+a problem or would like to suggest a feature. We always appreciate a well-written,
+thorough issue description. :brain:
+
+Some points to consider when opening an issue:
+- **Fully complete the provided issue template.** The bug report and feature request templates
+    specify all the information we need to quickly and efficiently address your issue. Be clear,
+    concise, and descriptive.
+    Provide as much information as you can, including steps to reproduce, stack traces, etc.
+- **Use [GitHub-flavored Markdown](https://help.github.com/en/github/writing-on-github/basic-writing-and-formatting-syntax).**
+    Especially put code blocks and console outputs in backticks (```). This improves readability.
+- **Do not open a duplicate feature request.** Search for existing feature requests first. If you
```

Comment:
This should be the 2nd point after checking release version.

Reply:
done